cmd/argo/commands/root.go

```diff
@@ -141,9 +141,11 @@ If your server is behind an ingress with a path (you'll be running "argo server
 	viper.AutomaticEnv()
 	viper.SetEnvPrefix("ARGO")
 	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_", ".", "_"))
+	// bind flags to env vars (https://github.com/spf13/viper/tree/v1.17.0#working-with-flags)
 	if err := viper.BindPFlags(command.PersistentFlags()); err != nil {
```

Comment:
Wouldn't this remove structured key configuration from viper?
E.G:
```yaml
foo:
     bar

```
 Would result in being saved as in .yaml configuration:
 ```yaml
 bar
```

Reply:
Are you referring to the fact that `PersistentFlags` is being used?

We don't use structured key configuration and we _don't_ want structured/nested env vars either. So that would be intended behavior.

Per the PR description, this is only intended to be used by `argo server`, but in this case the flag is at the root level.

Reply:
Ah okay, i was having the same issues in my project,  I use  structured/nested env vars, which unfortunately does not work with `BindPFlags`, so i was just really curious.  I really hope this will get  resolved in v2.